CONTRIBUTING.md

```diff
@@ -0,0 +1,316 @@
+At a high level, the contents of the commit message should be to convey the rationale of the change, without delving into much detail.
+For example, “bits were not set right” leaves the reviewer wondering about which bits and why they weren’t “right”.
+In contrast, “Correctly compute ‘is dependent type’ bits in ‘Type’” conveys almost all there is to the change.
+
+Below are some guidelines about the format of the commit message itself:
+
+- Separate the commit message into a single-line title and a separate body that describes the change.
+- Make the title concise to be easily read within a commit log and to fit in the subject line of a commit email.
+- In changes that are restricted to a specific part of the code, include a [tag] at the start of the line in square brackets—for example, “[stdlib] …” or “[SILGen] …”.
+This tag helps email filters and searches for post-commit reviews.
+- When there is a body, separate it from the title by an empty line.
+- Make body concise, while including the complete reasoning.
+Unless required to understand the change, additional code examples or other details should be left to bug comments or the mailing list.
+- If the commit fixes an issue in the bug tracking system, include a link to the issue in the message.
+- For text formatting and spelling, follow the same rules as documentation and in-code comments—for example, the use of capitalization and periods.
+- If the commit is a bug fix on top of another recently committed change, or a revert or reapply of a patch, include the Git revision number of the prior related commit, for example “Revert abcdef because it caused bug#”.
+
+For minor violations of these guidelines, the community normally favors reminding the contributor of this policy over reverting.
+Minor corrections and omissions can be handled by sending a reply to the commits mailing list.
+
+### Attribution of Changes
+
+When contributors submit a change to a Swift subproject, after that change is approved, other developers with commit access may commit it for the author.
+When doing so, it is important to retain correct attribution of the contribution. Generally speaking, Git handles attribution automatically.
+
+We do not want the source code to be littered with random attributions like “this code written by J. Random Hacker”, which is noisy and distracting.
+Do not add contributor names to the source code or documentation.
+
+In addition, don’t commit changes authored by others unless they have submitted the change to the project or you have been authorized to submit on their behalf—for example, you work together and your company authorized you to contribute the changes.
+The author should first either submit the change through a pull request to the relevant project, email the development list, or add a bug tracker item.
+If someone sends you a change privately, encourage them to submit it to the appropriate list first.
```

Comment:
I don’t know how ancient this paragraph it but most of it isn’t relevant with git and GitHub. I think the only relevant part is

```md
Don’t commit changes authored by others unless they have submitted the change to the project or you have been authorized to submit on their behalf — for example, you work together and your company authorized you to contribute the changes.
```

Reply:
removed the non-github pieces and left in the "don't submit the work of others" bit with the caveat